client/doc.go

```diff
@@ -0,0 +1,156 @@
+VRF key). We assume a separate PKI exists to manage the server's keys.
+
+Use of a CONIKS account verification bot is optional (see
+https://godoc.org/github.com/coniks-sys/coniks-go/bots for details).
+If the corresponding CONIKS key server requires that all registered names
+be validated by a CONIKS registration bot, the client also pins the
+registration bot that proxies all of its registration requests.
+Additionally, as the network of CONIKS auditors is currently very small,
+the client also maintains a list of pinned auditors. In a future release,
+we plan to introduce an automatic update mechanism for the auditors list
+while the development of a more scalable auditor discovery mechanism
+is underway.
+
+Protocol:
+- The user enters her username into the CONIKS client GUI prompt.
+
+- The client generates a new public-private key pair for this username,
+and stores the keys in secure? local storage on the device.
+
+- The client sends a registration request
+        reg_req = (username, key)
+to the server, or to the account validation bot, if used. See
+https://godoc.org/github.com/coniks-sys/coniks-go/bots for the
+CONIKS account validation protocol specification.
+
+- If the registration request is accepted*, the server returns a
+registration proof.
+More specifically, the server returns a proof of absence:
+        reg_pf = (auth_path, str)
+If the server uses the temporary binding protocol extension, reg_pf will
+also include a temporary binding (TB). See
+https://godoc.org/github.com/coniks-sys/coniks-go/protocol/extensions/tb
+for the temporary binding protocol extension specification.
+
+- The client audits the received STR including a hash chain check.
+See https://godoc.org/github.com/coniks-sys/coniks-auditor for the
+auditing protocol specification. *Note: Auditing is not implemented yet.*
+
+- Assuming the audit passes, the client proceeds to verify the TB, if
+included: it verifies the server's signature on th TB and checks that
+they public key in the TB matches the public key it sent to the proxy
+in reg_req.
+
+- Next, the client verifies that auth_path is a proof of absence by
+checking that its private index is invalid for the username, and its
+commitment is invalid for the public key it registered in reg_req.
+The client then recomputes the server's directory root using auth_path,
+and verifies that str includes the recomputed root in its signature.
+
+- If the registration proof is invalid, the client notifies the user. The
+Developing the reporting mechanism is planned for a future release.
+
+* The registration request may be denied for one of the following reasons:
+(1) The client attempts to register a name that already exists in the
+CONIKS key directory (ErrorNameExisted); the server returns a
+privacy-preserving proof of inclusion in its response. (2) The server
+encounters an internal error when attempting to register the name
+(ErrorDirectory).
```

Comment:
So, because this documentation is meant to describe the client-side protocols at a lower level for developers, I think it's perfectly reasonable to include relevant implementation details, such as the fact that a registration request won't succeed if the server encounters some internal error. 

Maybe "denied" isn't the right term to use in this case, but rather use "success/failure" terminology?